### PR TITLE
Cmake3 31

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will update to use cmake 3.31,   update the toml library and remove remnants of json_cpp

### Proposed changes
-update toml11 version in use
- require clang15 for builds with C++20 (still works with older version using C++17)
- resolve several warnings
- try to speed up the azure windows builds

### CODE REVIEW

File: docs/user-guide/installation/mac.md [hunk 1]
```diff
@@ -2,10 +2,11 @@
...
 ## Requirements
 
-- C++17 compiler
-- CMake 3.10 or newer (if using clang with libc++, use 3.18+)
+- C++20 compiler
+- CMake 3.22 or higher
+- XCode 14 or higher
 - git
-- Boost 1.67 or newer
+- Boost 1.75 or newer
```
Comment (nightlark) [inline docs/user-guide/installation/mac.md:9]: Why is it higher on macOS?

File: docs/user-guide/installation/windows.md [hunk 2]
```diff
@@ -18,16 +18,16 @@ _Note_: Keep in mind that your CMake version should be newer than the boost vers
 
 ### Set up your Environment
 
-**1.** Install Microsoft Visual C++ 2019 or newer (2017 may work, but is no longer tested by CI builds) [MSVC](https://visualstudio.microsoft.com/)
+**1.** Install Microsoft Visual C++ 2019 or newer that supports C++20 [MSVC](https://visualstudio.microsoft.com/)
 
-**2.** Install [Boost](https://dl.bintray.com/boostorg/release/1.74.0/binaries/)
-1.67 or later. For CMake to detect it automatically either
+**2.** Install [Boost](https://dl.bintray.com/boostorg/release/1.84.0/binaries/)
+1.74 or later. For CMake to detect it automatically either
```
Comment (nightlark) [inline docs/user-guide/installation/windows.md:24]: Earlier it says Boost 1.75 or newer -- is the minimum not 1.73 like it is with Linux?

File: src/helics/shared_api_library/helicsApps.h [hunk 3]
```diff
@@ -49,18 +49,15 @@ time, the app will not likely function normally; other query, information calls,
 HELICS_EXPORT HelicsFederate helicsAppGetFederate(HelicsApp app, HelicsError* err);
 
 /**
- * Create a HelicsApp object.
+ * Load a file to an App
```
Comment (nightlark) [inline src/helics/shared_api_library/helicsApps.h:52]:
```suggestion
 * Load a file to an App.
```

File: src/helics/shared_api_library/helicsApps.h [hunk 4]
```diff
@@ -49,18 +49,15 @@ time, the app will not likely function normally; other query, information calls,
...
  *
- * @details Create a HelicsApp object.
+ * @details load a configuration file for an app
```
Comment (nightlark) [inline src/helics/shared_api_library/helicsApps.h:54]:
```suggestion
 * @details Loads a configuration file for an app.
```

File: src/helics/shared_api_library/helicsApps.h [hunk 5]
```diff
@@ -49,18 +49,15 @@ time, the app will not likely function normally; other query, information calls,
...
  *
  * @param appName A string with the name of the app, can be NULL or an empty string to pull the default name from fedInfo or the config
  * file.
- * @param appType The type of app to create.
+ * @param app The app to load a file
```
Comment (nightlark) [inline src/helics/shared_api_library/helicsApps.h:58]:
```suggestion
 * @param app The app to load a file.
```

File: docs/user-guide/installation/mac.md (hunk 1 above)
Comment (phlptp) [reply docs/user-guide/installation/mac.md:9]: likely a mistype

Comment (phlptp) [inline docs/user-guide/installation/mac.md:9]:
```suggestion
- Boost 1.73 or newer
```

File: docs/user-guide/installation/windows.md [hunk 6]
```diff
@@ -6,10 +6,10 @@ Windows installers are available with the different [releases](https://github.co
 
 ## Build Requirements
 
-- Microsoft Visual C++ 2019 or newer (MS Build Tools also works; VC++ 2017 may work but is no longer tested by CI builds)
-- CMake 3.14 or newer (CMake should be newer than the Visual Studio and Boost version you are using; if using clang with libc++ use 3.18+)
+- Microsoft Visual C++ 2019 or newer (MS Build Tools also works)
+- CMake 3.22 or newer (CMake should be newer than the Visual Studio and Boost version you are using;)
 - git
-- Boost 1.67 or newer
+- Boost 1.75 or newer
```
Comment (phlptp) [inline docs/user-guide/installation/windows.md:12]:
```suggestion
- Boost 1.75 or newer (needs to have boost cmake configuration)- 1.73 and 1.74 may work with older versions of Cmake
```

File: docs/user-guide/installation/windows.md (hunk 2 above)
Comment (phlptp) [inline docs/user-guide/installation/windows.md:24]:
```suggestion
1.75 or later. For CMake to detect it automatically either
```

**Pull request merged**